### PR TITLE
Refactoring scorer + update_metrics

### PR DESCRIPTION
**Pull request by pruksmhc**
Refactoring scorers to make it more flexible. 
Changes: 
-> Added update_metrics to Task class to allow for flexibility in scorers (moving that logic out of src/models.py to individual tasks) 
-> Added self.subset_scorers to tasks to fit API 

Currently only affects single single sentence tasks and pair sentence classification tasks (non-diagnostic)
Current status:
-Tested on WNLI, STSB, COLA
-Testing on the rest of the affected tasks

### CODE REVIEW

File: src/tasks/tasks.py [hunk 1]
```diff
@@ -426,6 +435,12 @@ def get_metrics(self, reset=False):
         return {'mcc': self.scorer1.get_metric(reset),
                 'accuracy': self.scorer2.get_metric(reset)}
 
+    def update_metrics(self, logits, labels, tagmask=None):
+        logits, labels = logits.detach(), labels.detach()
```
Comment (sleepinyourhat) [inline src/tasks/tasks.py]: Why the detach? (Don't think it's wrong, just not sure.)

File: src/tasks/tasks.py [hunk 2]
```diff
@@ -437,6 +452,7 @@ def __init__(self, path, max_seq_len, name, **kw):
         self.val_metric_decreases = False
         self.scorer1 = Correlation("matthews")
         self.scorer2 = CategoricalAccuracy()
+        self.scorers =[self.scorer1, self.scorer2]
```
Comment (sleepinyourhat) [inline src/tasks/tasks.py]: autopep8

**Comment by iftenney** [comment]
Is this intended to only work for single-sentence tasks? It seems fairly safe and general enough to do everything.

File: src/tasks/tasks.py (hunk 1 above)
Comment (HaokunLiu) [reply src/tasks/tasks.py]: Pytorch sometimes reports errors when converting a gpu tensor to cpu, if it is needed to compute some other tensor that requires grad. Here, logits and labels are used to compute loss, while they will be moved to cpu within the scorer updating function.

**Comment by sleepinyourhat** [comment]
Thanks!
